pkg/appender/lru_cache.go

```diff
@@ -35,21 +35,15 @@ func NewCache(max int) *Cache {
 func (c *Cache) Add(key uint64, value *MetricState) {
 	c.mtx.Lock()
 	defer c.mtx.Unlock()
-	if ee, ok := c.cache[key]; ok {
-		c.free.Remove(ee)
-		//check if element was already in list and if not push to front
-		c.used.MoveToFront(ee)
-		if c.used.Front() != ee {
-			c.used.PushFront(ee)
-		}
-		ee.Value.(*entry).value = value
-		return
+	if ele, ok := c.cache[key]; ok {
+		c.free.Remove(ele)
+		c.used.Remove(ele)
```

Comment:
In my opinion better add a comment explaining why you delete from both lists, and that you expect one to "fail"